Cake.Frosting.Issues.Recipe/Cake.Frosting.Issues.Recipe/Context/State/IssuesState.cs

```diff
@@ -56,6 +62,17 @@ public class IssuesState : IIssuesState
     /// <inheritdoc />
     public IList<(IIssueProvider, string)> IssueProvidersAndRuns => this.issueProvidersAndRuns.AsReadOnly();
 
+    /// <summary>
+    /// Gets the default project root directory.
+    /// This method can be overridden in derived classes to customize the project root directory calculation.
+    /// Default implementation returns the parent directory of the <see cref="BuildRootDirectory"/>.
+    /// </summary>
+    /// <returns>The project root directory.</returns>
+    protected virtual DirectoryPath GetProjectRootDirectory()
```

Comment:
Rename to `DetermineProjectRootDirectory` to be consistent with other methods and avoid confusion with `ProjectRootDirectory` property.